### PR TITLE
capture(cdc): put OwnerRevision into context

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- The new scheduler needs to get the `ownerRev` of the owner, which is a number monotonically increasing cluster-wise, marking the order in which owners are elected.

### What is changed and how it works?
- Put `OwnerRevision` into `context`.
- Added method `GetOwnerRevision` in our wrapped `CDCEtcdClient`.
- Unit tests.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by amyangfei — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

File: cdc/capture/capture.go [hunk 1]
```diff
@@ -283,10 +283,28 @@ func (c *Capture) campaignOwner(ctx cdcContext.Context) error {
...
+		if err != nil {
+			if errors.Cause(err) == context.Canceled {
+				return nil
+			}
+			return errors.Trace(err)
+		}
+
+		// We do a copy of the globalVars here to avoid
+		// accidental modifications and potential race conditions.
+		globalVars := *ctx.GlobalVars()
```
Comment (lonng) [inline cdc/capture/capture.go:296]: The struct `GlobalVars` share some pointer fields. Do we need to implement the deep copy for the `GlobalVars`?

Comment (liuzix) [reply cdc/capture/capture.go:296]: I don't think so. Those are globally shared resources such as the pd client, the tikv client, etc, and they do not need to be copied.

**Pull request merged**